src/field/vecops.cxx

```diff
@@ -35,43 +35,42 @@
  **************************************************************************/
 
 const Vector2D Grad(const Field2D &f, CELL_LOC outloc) {
-  Vector2D result(f.getMesh());
-
   TRACE("Grad( Field2D )");
-  
-  if (outloc == CELL_DEFAULT) {
-    outloc = f.getLocation();
-  }
+
+  Vector2D result(f.getMesh());
 
   result.x = DDX(f, outloc);
   result.y = DDY(f, outloc);
   result.z = DDZ(f, outloc);
```

Comment:
Vector2D can indeed have VSHIFT but there is no handling of it anywhere currently.

We could just use `result.setLocation(result.x.getLocation());` to set the location at the end in the meantime. Currently VSHIFT isn't supported for this routine (it should throw if you try) -- changing this may be regarded as a bug fix or it may be a feature!

Reply:
I agree about trying to get rid of the three argument versions.